lib/App/Config/Attribute.pm

```diff
@@ -0,0 +1,103 @@
+package App::Config::Attribute;
+
+use Moose;
+extends 'App::Config::Node';
+use namespace::autoclean;
+use JSON::XS qw( decode_json );
+
+use MooseX::Types -declare => ['LongStr'];
+use Moose::Util::TypeConstraints;
+use Try::Tiny;
+
+subtype 'LongStr', as 'Str';
+
+=head1 NAME
+
+App::Config::Attribute
+
+=head1 ATTRIBUTES
+
+=cut
+
+has 'version' => (
+    is => 'rw',
+);
+
+sub _check_type {
+    my ($self, $value) = @_;
+    my $def = $self->{definition};
+    $self->{_json_string} //= $def->{isa} eq 'json_string' ? 1 : 0;
+    if ($self->{_json_string}) {
+        try { $value = decode_json($value) } catch { die "Couldn't decode JSON attribute" };
```

Comment:
Let's be more specific here :
` die "Couldn't decode JSON attribute $value : $_"`